crates/sui-rosetta/src/operations.rs

```diff
@@ -138,12 +123,57 @@ impl Operation {
         operations
     }
 
-    pub async fn parse_transaction_data(
+    /// Parse operation input from rosetta to Sui transaction
+    pub async fn create_data(
         operations: Vec<Operation>,
         metadata: ConstructionMetadata,
     ) -> Result<TransactionData, Error> {
-        let action: SuiAction = operations.try_into()?;
-        action.try_into_data(metadata).await
+        // Currently only PaySui is support,
+        // first operation is PaySui operation and second operation is the budget operation.
+        if operations.len() != 2 || operations[0].type_ != OperationType::PaySui {
+            return Err(Error::new_with_msg(
+                ErrorType::InvalidInput,
+                "Malformed operation.",
+            ));
+        }
+        let pay_sui_op = &operations[0];
+        let budget_op = &operations[1];
+
+        let account = pay_sui_op
+            .account
+            .as_ref()
+            .ok_or_else(|| Error::missing_input("operation.account"))?;
+        let address = account.address;
+        let pay_sui = pay_sui_op
+            .metadata
+            .clone()
+            .ok_or_else(|| Error::missing_input("operation.metadata"))?;
+        let pay_sui: PaySuiMetadata = serde_json::from_value(pay_sui)
+            .map_err(|e| Error::new_with_cause(ErrorType::MalformedOperationError, e))?;
+        let gas = metadata.sender_coins[0];
+        let budget_value = budget_op
+            .metadata
+            .clone()
+            .and_then(|v| v.pointer("/budget").cloned())
+            .ok_or_else(|| Error::missing_input("gas budget"))?;
+        let budget = budget_value
+            .as_u64()
+            .or_else(|| budget_value.as_str().and_then(|s| u64::from_str(s).ok()))
+            .ok_or_else(|| {
+                Error::new_with_msg(
+                    ErrorType::InvalidInput,
+                    format!("Cannot parse gas budget : [{budget_value}]").as_str(),
+                )
+            })?;
+
+        Ok(TransactionData::new_pay_sui(
```

Comment:
would be great if you can take a look at https://github.com/MystenLabs/sui/pull/5597 as well, which will do a validity check there, thanks!